### ISSUE TITLE
Communicate what repos have precise intel available

### ISSUE THREAD

**Issue by jjinnii** [sourcegraph/sourcegraph#31180]
#### Feature request description

Communicate to users when a repo has precise intel available. While the precise intel banner in the code intel popover generally promotes precise intelligence, we don't communicate when precise intel is available for repos and/or file that users are interested in.  The precise/search-based tags communicate what the user has currently set up, not whether precise intel is available.

The goal of this feature is to promote the discoverability of precise code intel.

#### Designs

- [Repo page label ](https://www.figma.com/file/p5rVseXjNvXRkvXCdttbWt/Semantic-badge?node-id=587%3A6780)
- Need to explore labels for file and/or search results page. 

#### Additional context

Iterated from [Rob's designs](https://www.figma.com/file/p5rVseXjNvXRkvXCdttbWt/Semantic-badge?node-id=0%3A1).

**Comment by jjinnii** [comment]
@macraig I don't think I've actually vocalized this before, so would love your take on what I wrote under popover! Do you think it makes sense to loop in the popover banner improvement work into this feature? 

My thinking was, I could easily make the current search-based banner smaller/nicer, and it would solve for the core user problem of the banner being intrusive. I think it's a question of: can we make it _more effective_ in promoting precise intel, which is actually the goal of that search-based banner in the first place. (just a bit of a roundabout way of doing so. We're essentially saying "this is search-based! Also precise exists" in the current banner)

**Comment by macraig** [comment]
@jjinnii big +1 to looping in the popover banner improvement. As you said both issues are related to the same theme and goal, and it would be awesome to tackle them as part of the solution to the same problem!

**Comment by mrnugget** [comment]
> * **File page**: The file view is a core view for the user, and I'm hesitant to place a sticky badge for precise intel within the file view. ([Rob's design](https://www.figma.com/file/p5rVseXjNvXRkvXCdttbWt/Semantic-badge?node-id=681%3A9320) for reference) The popover banner helps to promote code intel when a user is in a file and using code intel; adding a badge to the file view risks disrupting code readability and becoming intrusive to the user. I can conduct user testing if the team feels this should be explored further.

I strongly disagree with this. The file view is *the* place where I'd want to know whether precise/search or any other form of code intel is available before maniacally wiggling the cursor to find out whether I'll get a hover tooltip or not. Our product is so unpredictable in this regard that I'm constantly wondering why I don't get a tooltip.

**Comment by jjinnii** [comment]
@macraig  From[ today's sync](https://docs.google.com/document/d/1R4gXavKwajVRplHSy1ECn_ZHMoQZIwiGKqWWb2SdbUE/edit#bookmark=id.hj6ix7afx9il), there is a need to re-align on the goals of the code intel badge work. As mentioned in the issue, I am happy to design an option and test the UX for a file code intel badge. However, we seem to have different ideas on what the goal of this feature is, which would impact my design for the badges in all locations.  

My understanding was that the primary goal of this feature is to promote the discoverability of precise code intel and communicate to the user when precise intel is _available_. 

Today's sync surfaced a few other potential goals (@mrnugget please add any other/correct the below): 
- Communicate when both search-based _and_ precise intel is available 
- Communicate when code intel is available without the user having to click/hover in a file 
- Communicate when errors or latency occur due to code intel indexing/uploading

@chrismwendt shared what used to be a [code intel status indicator](https://github.com/sourcegraph/sourcegraph/issues/788) that existed in the past. This is a compelling design option if the primary goal of the badge work is to communicate the status of code intelligence.

As a next step, we should come to an agreement on whether the goal of the badge is discoverability or informational. I'm not saying one design can't achieve both, but it helps drive the design and decision-making process to have a clear shared understanding of what goal, thus the definition of success, is driving us to introduce these changes. (e.g. is it the # of people who adopt precise intel? Is it reduced number of complaints/issues related to not knowing what code intel was available?)

**Comment by mrnugget** [comment]
Those goals look correct to me. The overarching goal is this point in our [strategy document](https://handbook.sourcegraph.com/strategy-goals/strategy/code-graph/code-intelligence/#themes):

> Make code navigation feel fast and reliable

Keyword is _reliable_, which, I think, also means making code navigation _predictable_ (because how can you rely on something if you can't predict what it'll do?).

We've had a lot of discussions (and tickets and Slack threads) over the years on how users have a hard time anticipating _if_ and _when_ a hover tooltip will show and whether it will contain useful information. Examples:

- https://github.com/sourcegraph/sourcegraph/issues/26015
- https://github.com/sourcegraph/sourcegraph/issues/19034
- https://github.com/sourcegraph/sourcegraph/issues/18018
- https://github.com/sourcegraph/sourcegraph/issues/12525
- https://github.com/sourcegraph/sourcegraph/issues/9385
- https://github.com/sourcegraph/sourcegraph/issues/8830

That's just from a "hover tooltip" search in our issues. Yes, these are all slightly different glitches/bugs/problems but there is one underlying theme: the users don't know what to expect or why things happened the way they did. Unsure whether tooltip didn't show up because of network problems, or because there never will be hover tooltip on that token (some tokens don't have hovers and that's completely fine), or because no data is available (search-based is indexing in background, no LSIF dump uploaded), etc.

So when we planned the work for the Figma files and wrote it down in https://github.com/sourcegraph/sourcegraph/issues/31071 we didn't have _this ticket here_, because it didn't exist yet. We looked at Figma and decided that we can do all of that (tell the user whether it's possible to add precise code intel to the repo/file/etc. they're looking at) _and_ also tell them when we do have precise or search based code intel and which data it used and how old that data is etc. You can see in the pseudo-code sketched out API in #31071 what info is available.

Essentially: it's not hard to add more, higher-fidelity data to the API, so let's do that and fix both problems at once: nudge the user to upload data if there's no data and also make the product predictable/reliable for users.

I also don't think these goals are opposed and don't see why we can't have a badge that solves both of these issues.

(cc @efritz and @Strum355 since it's Noah's ticket and both of them are involved)

**Comment by macraig** [comment]
@jjinnii @mrnugget what I'm hearing is that we have two distinct themes/goals here. One is "Make code intel more discoverable" (this goal is oriented to grow code intel awareness) and the other is "Make code intel more reliable" which would improve predictability and quality of life. The original goal for the badge project was the first, to add CTAs or improve the ones we have (like the banner). 

My suggestion: Let's keep this issue to track discoverability-oriented work (scope already defined above) and create a new "Make code intel reliable" themed issue tracker that includes the context and issues @mrnugget added above, with the aim of communicating code intel status in a clearer way.

Treating them as separate user problems will help us arrive at the best possible solutions, and we might even find that we have the same solution for both themes (or not, both are fine as long as we solve the root problems).

**Comment by mrnugget** [comment]
Agree, it's two separate problems. My point was that I think we can solve both at the same time with the same design and API. I want us to avoid a scenario where we build a solution for problem 1, then build a solution for problem 2, realise "oh, wait, it's the same thing", and then having to re-build both solutions.

**Comment by macraig** [comment]
In the case you describe above wouldn't it be more of an iterative approach rather than re-building? For example, we implement the discoverability badge and move on to solve predictability. If the solution is building on the same design and API, wouldn't we be leveraging our previous work and building on top of it instead of re-building both?

**Comment by mrnugget** [comment]
Hehe, yeah, sure. I don't think re-building is opposed to iterative :) But as I wrote [here](https://github.com/sourcegraph/sourcegraph/issues/31180#issuecomment-1054372259) I fear the goals might be opposed and not complementary. Example: you don't want to shove "hey, use indexers to get precise code intel" down user's throats on every screen, but "will I get code intel if I hover over something?" is information that is great to have on every code screen.

That being said: these are non-blocking concerns. Let's start designing/building/testing something and then we'll know much more.

**Comment by macraig** [comment]
FYI I created a separate issue for predictability issues in https://github.com/sourcegraph/sourcegraph/issues/32293
